core/src/connection/util.rs

```diff
@@ -0,0 +1,38 @@
+use std::{io, task::{Poll, Context}};
+
+use multiaddr::Multiaddr;
+
+use crate::muxing::StreamMuxerBox;
+
+use super::{ConnectionHandler, Substream, SubstreamEndpoint, ConnectionHandlerEvent};
+
+
+#[derive(Debug)]
+pub struct TestHandler();
```

Comment:
Instead of making this handler public, can we move the test you added to `tests/` and hence reuse the handler from there? I don't think we should be adding this one to the public about of `libp2p-core`.

Reply:
To resolve the issue about `Pool` being `pub(crate)`, I'd suggest to write the unit test against the `Network` component (which is the public API of `libp2p-core`).

The structure should roughly be the same:

1. Make a new `Network`
2. `dial` a peer
3. Retrieve it with `peer`
4. Convert it into a `DialingPeer`
5. `disconnect` the `DialingPeer`
6. Observe the aborted event

Reply:
Oh! good idea, thanks!